src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2046,6 +2086,26 @@ bool tryDecodeExtensionType(out bool isExplicit, [NotNullWhen(true)] out TypeSym
                 Debug.Assert(underlyingType is not null);
                 return true;
             }
+
+            bool validateUnderlyingInstanceField(FieldDefinitionHandle underlyingInstanceFieldHandle, PEModuleSymbol moduleSymbol, TypeSymbol underlyingType)
```

Comment:
There's two reasons to care about the presence of the field:
1. for `Unsafe.As` to be safe, we need the extension to be properly formed. We could do lighter validation (just the type) but I don't see the point to accept ill-formed extension types
2. `this.UnderlyingMember` in an instance extension type member will need to be rewritten to access this field, something like `this.<UnderlyingInstanceField>$.UnderlyingMember` (this will be in a later PR)

Reply:
You're correct. That leaves reason 1